tagging.tf

```diff
@@ -17,8 +17,15 @@ resource "null_resource" "tags" {
 
   # Code found here https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-using-tags#code-try-22
   provisioner "local-exec" {
-    interpreter = ["/bin/bash", "-c"]
-    command = templatefile("${path.module}/script/tag.sh.tmpl",
+    interpreter = var.interpreter == "PowerShell" ? ["PowerShell"] : ["/bin/bash", "-c"]
+    command     = var.interpreter == "PowerShell" ? templatefile("${path.module}/script/tag.ps1.tmpl",
+      {
+        behavior        = var.behavior
+        resource_id     = var.resource_ids[count.index]
+        tags            = local.tags
+        subscription_id = data.azurerm_client_config.current.subscription_id
+      }
+    ) : templatefile("${path.module}/script/tag.sh.tmpl",
```

Comment:
Also, it would be better to have a `is_powershell` local, no ?

```suggestion
    command     = templatefile(format("%s/script/%s", path.module, local.is_powershell ? "tag.ps1.tmpl" : "tag.sh.tmpl"),
```

Reply:
I love the use of format for a simpler, cleaner conditional!  Updating my PR to reflect that.

For the local I don't think this is something that should be baked into the module itself, because PowerShell can run on non-Windows hosts (and even on Windows hosts there is always the chance some other interpreter could be used).  As a result the file path test doesn't always correspond 1-to-1 with the interpreter.  Such a local could be a useful thing to include in the project that is consuming the module (example below), but the module shouldn't make the assumption on behalf of the project.

```hcl
# project_main.tf
locals {
  # Directories start with "C:..." on Windows; All other OSs use "/" for root.
  detected_interpreter = substr(pathexpand("~"), 0, 1) == "/" ? "PowerShell" : "bash"
}

# invoke module, passing local to interpeter var
module "tagging" {
  source  = "claranet/tagging/azurerm"
  version = "4.0.2"

  interpreter  = local.detected_interpreter
  nb_resources = 2
  resource_ids = [var.myresource_id_1, var.myresource_id_2]
  behavior     = "merge" # Must be "merge" or "overwrite"

  tags = {
    "foo"        = "bar"
    "monitoring" = "true"
  }
}

```

Reply:
I agree with you, I was thinking something a lot simpler like 

```
locals {
    is_powershell = var.interpreter == "PowerShell"
}
```

:smiley:

Reply:
Oh sure, that would work just fine.  Forgive my ignorance--what is the purpose of having the variable passed to a local before being consumed in a resource?

Reply:
Makes sense, and would certainly help avoid some spaghetti code I've ~~written~~ seen in the past.  Thanks for clarifying!